src/CommitmentsV2/SFA.DAS.CommitmentsV2.UnitTests/Services/EditValidation/OverlapValidationTests.cs

```diff
@@ -1,58 +1,91 @@
 using System.Linq;
 using System.Threading.Tasks;
 
 namespace SFA.DAS.CommitmentsV2.UnitTests.Services.EditValidation
 {
     public class OverlapValidationTests
     {
-        [Test]
-        public async Task When_StartDate_Overlaps()
+        [TestCase(Party.Employer)]
+        [TestCase(Party.Provider)]
+        public async Task When_StartDate_Overlaps(Party party)
         {
             var fixture = new EditApprenticeshipValidationServiceTestsFixture();
-            fixture.SetupMockContextApprenticeship().SetupOverlapService(true, false);
-            var request = fixture.CreateValidationRequest(employerRef: "abc");
+            var tupleResult = await SetupAuthenticationContext(party, fixture, true, false);
 
-            var result = await fixture.Validate(request);
+            var result = tupleResult.Result;
+            var errorText = tupleResult.ErrorText;
 
             Assert.NotNull(result.Errors);
             Assert.AreEqual(1, result.Errors.Count);
-            Assert.AreEqual("The date overlaps with existing training dates for the same apprentice. Please check the date - contact your training provider for help", result.Errors[0].ErrorMessage);
+            Assert.AreEqual(errorText, result.Errors[0].ErrorMessage);
             Assert.AreEqual("StartDate", result.Errors[0].PropertyName);
         }
 
-        [Test]
-        public async Task When_EndDate_Overlaps()
+        private static async Task<(EditApprenticeshipValidationResult Result, string ErrorText)> 
+            SetupAuthenticationContext(Party party, EditApprenticeshipValidationServiceTestsFixture fixture, bool startDateOverlap, bool endDateOverlap)
         {
-            var fixture = new EditApprenticeshipValidationServiceTestsFixture();
-            fixture.SetupMockContextApprenticeship().SetupOverlapService(false, true);
-            var request = fixture.CreateValidationRequest(employerRef: "abc");
+            var eaFixture = fixture.SetupMockContextApprenticeship()
+                .SetupOverlapService(startDateOverlap, endDateOverlap);
+            
+            var partyText = party == Party.Employer ? "training provider" : "employer";
+            var errorText = $"The date overlaps with existing training dates for the same apprentice. Please check the date - contact your {partyText} for help";
+
+            EditApprenticeshipValidationRequest request;
+
+            if (party == Party.Employer)
+            {
+                eaFixture.SetupAuthenticationContextAsEmployer();
+                request = eaFixture.CreateValidationRequest(employerRef: "abc");
+            }
+            else
+            {
+                eaFixture.SetupAuthenticationContextAsProvider();
+                request = eaFixture.CreateValidationRequest(providerRef: "abc");
+            }
 
             var result = await fixture.Validate(request);
 
+            return (result, errorText);
+        }
+
+        [TestCase(Party.Employer)]
```

Comment:
could we included the expected result error text as the second argument of the test case, I think this would make it clearer?

Reply:
good call, this will make the test simpler as well.
Done.